quic/s2n-quic-platform/src/io/turmoil.rs

```diff
@@ -80,26 +69,50 @@ impl Io {
             ));
         };
 
-        let rx_buffer = buffer::Buffer::new_with_mtu(max_mtu.into());
-        let mut rx = socket::Queue::new(rx_buffer);
+        let local_addr = socket.local_addr()?;
+        let local_addr: inet::SocketAddress = local_addr.into();
+        let payload_len: usize = max_mtu.into();
+        let payload_len = payload_len as u32;
+
+        let (rx, rx_producer) = {
+            let entries = 1024;
 
-        let tx_buffer = buffer::Buffer::new_with_mtu(max_mtu.into());
-        let tx = socket::Queue::new(tx_buffer);
+            let mut consumers = vec![];
 
-        let local_addr: SocketAddress = socket.local_addr()?.into();
+            let (producer, consumer) = ring::pair(entries, payload_len);
+            consumers.push(consumer);
 
-        // tell the queue the local address so it can fill it in on each message
-        rx.set_local_address(local_addr.into());
+            let rx = rx::Rx::new(consumers, max_mtu, local_addr.into());
 
-        let instance = Instance {
+            (rx, producer)
+        };
+
+        let (tx, tx_consumer) = {
+            let entries = 1024;
+
+            let mut producers = vec![];
+
+            let (producer, consumer) = ring::pair(entries, payload_len);
+            producers.push(producer);
+
+            let gso = crate::features::Gso::default();
+            gso.disable();
+            let tx = tx::Tx::new(producers, gso, max_mtu);
+
+            (tx, consumer)
+        };
+
+        tokio::spawn(run_io(socket, rx_producer, tx_consumer));
+
+        let el = EventLoop {
```

Comment:
nit: `el` feels a bit dense even though it has a super-local scope. Primarily because it makes me think of the [el rune](https://diablo.fandom.com/wiki/El_Rune) from Diablo II rather than "event loop", but perhaps others who didn't play Diablo II so much wouldn't have that problem 😅

Reply:
you say that like it's a bad thing :smile: